NGitLab.Mock/Config/GitLabHelpers.cs

```diff
@@ -0,0 +1,1279 @@
+        public static GitLabConfig SetDefaultUser(this GitLabConfig config, string username)
+        {
+            return Configure(config, _ =>
+            {
+                config.DefaultUser = username;
+            });
+        }
+
+        /// <summary>
+        /// Add an explicit group description in config
+        /// </summary>
+        /// <param name="config">Config.</param>
+        /// <param name="name">Name (required)</param>
+        /// <param name="configure">Configuration method</param>
+        public static GitLabConfig WithGroup(this GitLabConfig config, string name, Action<GitLabGroup> configure)
+        {
+            return Configure(config, _ =>
+            {
+                var group = new GitLabGroup
+                {
+                    Name = name ?? throw new ArgumentNullException(nameof(name)),
+                };
+
+                config.Groups.Add(group);
+                configure(group);
+            });
+        }
+
+        /// <summary>
+        /// Add an explicit group description in config
+        /// </summary>
+        /// <param name="config">Config.</param>
+        /// <param name="name">Name (required)</param>
+        /// <param name="id">Explicit ID (config increment)</param>
+        /// <param name="namespace">Parent namespace.</param>
+        /// <param name="description">Description.</param>
+        /// <param name="visibility">Visibility.</param>
+        /// <param name="defaultAsMaintainer">Define default user as maintainer.</param>
+        /// <param name="configure">Configuration method</param>
+        public static GitLabConfig WithGroup(this GitLabConfig config, string name, int id = default, string @namespace = null, string description = null, VisibilityLevel visibility = VisibilityLevel.Internal, bool defaultAsMaintainer = false, Action<GitLabGroup> configure = null)
+        {
+            return WithGroup(config, name, group =>
+            {
+                if (id != default)
+                    group.Id = id;
+
+                group.Namespace = @namespace;
+                group.Description = description;
+                group.Visibility = visibility;
+
+                if (defaultAsMaintainer)
+                {
+                    if (string.IsNullOrEmpty(group.Parent.DefaultUser))
+                        throw new InvalidOperationException("Default user not configured");
+
+                    WithUserPermission(group, group.Parent.DefaultUser, AccessLevel.Maintainer);
+                }
+
+                configure?.Invoke(group);
+            });
+        }
+
+        /// <summary>
+        /// Add a project description in config
+        /// </summary>
+        /// <param name="config">Config.</param>
+        /// <param name="name">Name (required)</param>
+        /// <param name="configure">Configuration method</param>
+        public static GitLabConfig WithProject(this GitLabConfig config, string name, Action<GitLabProject> configure)
```

Comment:
Is the name really mandatory? What if you are only interested in having an id in your tests. Maybe we should add another method:

````
public static GitLabConfig WithProject(this GitLabConfig config, int id, Action<GitLabProject> configure)
````